examples/erc20/lib.rs

```diff
@@ -97,71 +96,17 @@ pub mod my_erc20 {
         }
     }
     impl Erc20 for MyErc20 {}
-    impl IErc20 for MyErc20 {
-        #[ink(message)]
-        fn token_name(&self) -> Option<String> {
-            self._token_name()
-        }
-
-        #[ink(message)]
-        fn token_symbol(&self) -> Option<String> {
-            self._token_symbol()
-        }
-
-        #[ink(message)]
-        fn token_decimals(&self) -> u8 {
-            self._token_decimals()
-        }
-
-        #[ink(message)]
-        fn total_supply(&self) -> Balance {
-            self._total_supply()
-        }
 
-        #[ink(message)]
-        fn balance_of(&self, owner: AccountId) -> Balance {
-            self._balance_of(owner)
-        }
-
-        #[ink(message)]
-        fn transfer(&mut self, to: AccountId, value: Balance) -> Result<(), Erc20Error> {
-            panic_on_error(self._transfer(to, value))
-        }
-
-        #[ink(message)]
-        fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance {
-            self._allowance(owner, spender)
-        }
-
-        #[ink(message)]
-        fn transfer_from(&mut self, from: AccountId, to: AccountId, value: Balance) -> Result<(), Erc20Error> {
-            panic_on_error(self._transfer_from(from, to, value))
-        }
-
-        #[ink(message)]
-        fn approve(&mut self, spender: AccountId, value: Balance) -> Result<(), Erc20Error> {
-            panic_on_error(self._approve(spender, value))
-        }
-
-        #[ink(message)]
-        fn increase_allowance(&mut self, spender: AccountId, delta_value: Balance) -> Result<(), Erc20Error> {
-            panic_on_error(self._increase_allowance(spender, delta_value))
-        }
-
-        #[ink(message)]
-        fn decrease_allowance(&mut self, spender: AccountId, delta_value: Balance) -> Result<(), Erc20Error> {
-            panic_on_error(self._decrease_allowance(spender, delta_value))
-        }
-    }
+    make_trait!(MyErc20, IErc20);
```

Comment:
probably also a good idea to make `make_trait!` a derive macro and use like:
```rust
#[derive(IErc20)
struct MyErc20 {
  ...
}
```

Reply:
Ok, never mind, it would require a derive macro for each trait implementation. Let's stick with the way you did it here.

Reply:
Also it has another problem related to macro process ordering) So we can't use derive here=D